ios/FluentUI/People Picker/ContactView.swift

```diff
@@ -0,0 +1,189 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+
+open class ContactView: UIView {
+    private struct Constants {
+        static let contactWidth: CGFloat = 70.0
+        static let avatarViewWidth: CGFloat = 70.0
+        static let avatarViewHeight: CGFloat = 70.0
+        static let titleLabelMinimumHeight: CGFloat = 12.0
+        static let spacingBetweenAvatarAndFirstLabel: CGFloat = 13.0
+    }
+
+    @objc public var avatarImage: UIImage? {
+        didSet {
+            setupAvatarView(with: titleLabel?.text, and: subtitleLabel?.text, or: nil)
+        }
+    }
+    private let avatarView: AvatarView
+    private var titleLabel: UILabel?
+    private var subtitleLabel: UILabel?
+
+    /// Initializes the contact view by creating an avatar view with a first name and last name
+    ///
+    /// - Parameters:
+    ///   - title: String that will be the text of the top label
+    ///   - subtitle: String that will be the text of the bottom label
+    @objc public convenience init(title: String, subtitle: String) {
+        self.init(title: title, subtitle: subtitle, identifier: nil)
+    }
+
+    /// Initializes the contact view by creating an avatar view with an identifier
+    ///
+    /// - Parameters:
+    ///   - identifier: String that will be used to identify the contact (e.g. email, phone number, first name)
+    @objc public convenience init(identifier: String) {
+        self.init(title: nil, subtitle: nil, identifier: identifier)
+    }
+
+    private init(title: String?, subtitle: String?, identifier: String?) {
+        avatarView = AvatarView(avatarSize: .extraExtraLarge, withBorder: false, style: .circle)
+        super.init(frame: .zero)
+
+        if let title = title, let subtitle = subtitle {
+            subtitleLabel = UILabel()
+            setupAvatarView(with: title, and: subtitle, or: nil)
+            setupTitleLabel(using: title)
+            setupSubtitleLabel(using: subtitle)
+        } else if let identifier = identifier {
+            setupAvatarView(with: nil, and: nil, or: identifier)
+            setupTitleLabel(using: identifier)
+        }
+
+        backgroundColor = Colors.surfacePrimary
+        setupLayout()
+    }
+
+    public required init?(coder: NSCoder) {
+        preconditionFailure("init(coder:) has not been implemented")
+    }
+
+    private func setupAvatarView(with firstName: String?, and lastName: String?, or identifier: String?) {
```

Comment:
>setupAvatarView [](start = 17, length = 15)

I think we should just have 2 methods:
setupAvatarView(with firstName: String, and lastName: String)
setupAvatarView(with identifier: String)

there is no code shared between the cases where we have both title/subtitle and the case we have an identifier. Also, that way we don't need the params to be optional

Reply:
Agreed, the parameters look much cleaner done that way.